packages/contracts-bedrock/src/L1/ETHLockbox.sol

```diff
@@ -0,0 +1,165 @@
+    function paused() public view returns (bool) {
+        return superchainConfig.paused();
+    }
+
+    /// @notice Receives the ETH liquidity migrated from an authorized lockbox.
+    function receiveLiquidity() external payable {
+        if (!authorizedLockboxes[msg.sender]) revert ETHLockbox_Unauthorized();
+        emit LiquidityReceived(msg.sender);
+    }
+
+    /// @notice Locks ETH in the lockbox.
+    ///         Called by an authorized portal on a deposit to lock the ETH value.
+    function lockETH() external payable {
+        if (!authorizedPortals[msg.sender]) revert ETHLockbox_Unauthorized();
+        emit ETHLocked(msg.sender, msg.value);
+    }
+
+    /// @notice Unlocks ETH from the lockbox.
+    ///         Called by an authorized portal when finalizing a withdrawal that requires ETH.
+    ///         Cannot be called if the lockbox is paused.
+    /// @param _value The amount of ETH to unlock.
+    function unlockETH(uint256 _value) external {
+        if (paused()) revert ETHLockbox_Paused();
+        if (!authorizedPortals[msg.sender]) revert ETHLockbox_Unauthorized();
+        /// NOTE: Check l2Sender is not set to avoid this function to be called as a target on a withdrawal transaction
+        if (IOptimismPortal(payable(msg.sender)).l2Sender() != Constants.DEFAULT_L2_SENDER) {
+            revert ETHLockbox_NoWithdrawalTransactions();
+        }
+
+        // Using `donateETH` to avoid triggering a deposit
+        IOptimismPortal(payable(msg.sender)).donateETH{ value: _value }();
+        emit ETHUnlocked(msg.sender, _value);
+    }
+
+    /// @notice Authorizes an ETH lockbox to migrate its liquidity to the current ETH lockbox.
+    /// @param _lockbox The address of the ETH lockbox to authorize.
+    function authorizeLockbox(address _lockbox) external {
+        if (msg.sender != PAO()) revert ETHLockbox_Unauthorized();
+        if (!_samePAO(_lockbox)) revert ETHLockbox_DifferentPAO();
+        if (authorizedLockboxes[_lockbox]) revert ETHLockbox_AlreadyAuthorized();
+
+        authorizedLockboxes[_lockbox] = true;
+        emit LockboxAuthorized(_lockbox);
+    }
+
+    /// @notice Migrates liquidity from the current ETH lockbox to another.
+    /// @param _lockbox The address of the ETH lockbox to migrate liquidity to.
+    function migrateLiquidity(address _lockbox) external {
+        if (msg.sender != PAO()) revert ETHLockbox_Unauthorized();
+        if (!_samePAO(_lockbox)) revert ETHLockbox_DifferentPAO();
+
+        ETHLockbox(_lockbox).receiveLiquidity{ value: address(this).balance }();
+
+        emit LiquidityMigrated(_lockbox);
+    }
+
+    /// @notice Authorizes a portal to lock and unlock ETH.
+    /// @param _portal The address of the portal to authorize.
+    function _authorizePortal(address _portal) internal {
+        if (!_samePAO(_portal)) revert ETHLockbox_DifferentPAO();
+        if (authorizedPortals[_portal]) revert ETHLockbox_AlreadyAuthorized();
```

Comment:
We should be sure that this is the exact behavior that we want with `revert ETHLockbox_AlreadyAuthorized()`, because it will impact the way we reinit. With this revert, cannot pass in the same address more than once to `initialize`.

Reply:
Let's get rid of the check anyway